### PR TITLE
RAC-5938 : Refactor vagrant build groovy script

### PR DESCRIPTION
**Pull request by PengTian0**
Implement a groovy script to build vagrant box.
Test:
http://10.**.**.17*:8080/job/pengtian/job/vagrant/job/build

@panpan0000 @changev @HQebupt  @anhou @iceiilin

### CODE REVIEW

File: src/pipeline/rackhd/vagrant/Build.groovy [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package pipeline.rackhd.vagrant
+
+def buildBoxFromOVF(String target_dir, String cache_project_name, String os_ver, String rackhd_version, String rackhd_apt_repo, String rackhd_dir){
+    String cache_image_dir = "$WORKSPACE/cache_image"
+    // Copy ovf from the artifact of project VAGRANT_CACHE_BUILD
+    step ([$class: 'CopyArtifact',
```
Comment (panpan0000) [inline src/pipeline/rackhd/vagrant/Build.groovy]: I'm afraid it's better to put this (copy Artifact) from buildBoxFromOVF to caller (buildBox())
this function can focus on what it should do

File: src/pipeline/rackhd/vagrant/Build.groovy [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    String box_name = ""
+    def share_method = new pipeline.common.ShareMethod()
+    lock(label:label_name,quantity:1){
+        String node_name = share_method.occupyAvailableLockedResource(label_name, [])
+        node(node_name){
+            deleteDir()
+            String rackhd_dir = "RackHD"
+            String target_dir = "$WORKSPACE/vagrant"
+            // Checkout RackHD
+            share_method.checkout("https://github.com/RackHD/RackHD", rackhd_commit, rackhd_dir)
```
Comment (panpan0000) [inline src/pipeline/rackhd/vagrant/Build.groovy]: is it possible to make RackHD repo's URL/branch as a parameter ?

File: workflows/build_vagrant/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                "RACKHD_VERSION=${env.RACKHD_VERSION}",
+                "OS_VER=${env.OS_VER}",
+                "ARTIFACTORY_URL=${env.ARTIFACTORY_URL}",
+                "STAGE_REPO_NAME=${env.STAGE_REPO_NAME}",
+                "DEB_COMPONENT=${env.DEB_COMPONENT}",
+                "DEB_DISTRIBUTION=trusty",
+                "CACHE_PROJECT_NAME=VAGRANT_CACHE_BUILD"
+            ]){
+                def vagrant_builder = new pipeline.rackhd.vagrant.Build()
+                String rackhd_apt_repo = "${ARTIFACTORY_URL}/${STAGE_REPO_NAME} ${DEB_DISTRIBUTION} ${DEB_COMPONENT}"
```
Comment (panpan0000) [inline workflows/build_vagrant/Jenkinsfile]: Hmm... maybe we can just make those 4 param in Jenkins build into single 1 param :-) cry to tear...

**Review by panpan0000 — APPROVED** [review]

File: src/pipeline/rackhd/vagrant/Build.groovy (hunk 1 above)
Comment (PengTian0) [reply src/pipeline/rackhd/vagrant/Build.groovy]: Thanks!
Fixed.

File: src/pipeline/rackhd/vagrant/Build.groovy (hunk 2 above)
Comment (PengTian0) [reply src/pipeline/rackhd/vagrant/Build.groovy]: Thanks!
Fixed.

**Review by iceiilin — APPROVED** [review]

**Pull request merged**